reactioncommerce/reaction
# Better handling for unique routes for React Router

The old logic for handling route uniqueness was not sufficient in preventing duplicate routes in our new React Router world.

This PR attempts to correct this by handling route uniqueness by route path, and only after all routes have been had an opportunity to be processed.

**To Test:**

1. Add a duplicate route to a custom plugin register.js:

```
import { Reaction } from "/server/api";

Reaction.registerPackage({
  label: "Test",
  name: "test-plugin",
  icon: "fa fa-cubes",
  autoEnable: true,
  registry: [{
    route: "/product/:handle/:variantId?",
    name: "product",
    template: "test",
    workflow: "coreProductWorkflow"
  }]
});

```

2. Put this in any `.html` file in a client folder outside the imports directory" For example "/client/templates"

```
<template name="test">
  HELLO
</template>
```

3. Observe that your custom template overwrites the default PDP route.
